hw/hexagon/hexagon_dsp.c

```diff
@@ -139,11 +124,28 @@ static void hexagon_common_init(MachineState *machine, Rev_t rev,
 
         if (i == 0) {
             hexagon_init_bootstrap(machine, cpu);
+            if (!qdev_realize_and_unref(DEVICE(cpu), NULL, errp)) {
+                return;
+            }
```

Comment:
Somehow I assumed that this had to happen *after* other initialization things.  Not sure how I came to that conclusion though.

Does this more closely match downstream now?

Reply:
Yes, the order is now the same as downstream